src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs

```diff
@@ -146,6 +146,57 @@ internal static MsalTokenResponse CreateFromiOSBrokerResponse(Dictionary<string,
             return response;
         }
 
+        internal static MsalTokenResponse CreateFromAppProviderResponse(ExternalTokenResult externalTokenResponse)
+        {
+            ValidateExternalTokenResult(externalTokenResponse);
+
+            var response = new MsalTokenResponse
+            {
+                AccessToken = externalTokenResponse.RawAccessToken,
+                RefreshToken = null,
+                IdToken = null,
+                TokenType = BrokerResponseConst.Bearer,
+                ExpiresIn = externalTokenResponse.ExpiresInSeconds,
+                ClientInfo = null,
+                TokenSource = TokenSource.IdentityProvider,
+                TenantId = externalTokenResponse.TenantId
```

Comment:
If you leave this as `null`, I think MSAL will use the input tenant id by default, which is ok for client_credentials.